src/rc_string.c

```diff
@@ -65,29 +66,41 @@ static void rc_string_dispose(RC_STRING* content)
 
 static THANDLE(RC_STRING) rc_string_create_impl(const char* string)
 {
+    THANDLE(RC_STRING) result = NULL;
+
     /* Codes_SRS_RC_STRING_01_002: [ Otherwise, rc_string_create shall determine the length of string. ]*/
     size_t string_length = strlen(string);
-    size_t string_length_with_terminator = string_length + 1;
-
-    /* Codes_SRS_RC_STRING_01_003: [ rc_string_create shall allocate memory for the THANDLE(RC_STRING), ensuring all the bytes in string can be copied (including the zero terminator). ]*/
-    THANDLE(RC_STRING) temp_result = THANDLE_MALLOC_WITH_EXTRA_SIZE(RC_STRING)(rc_string_dispose, sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + string_length_with_terminator);
-    if (temp_result == NULL)
+    size_t memory_allocation_max_size = SIZE_MAX - (sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + 1);
+    if (string_length > memory_allocation_max_size)
     {
-        /* Codes_SRS_RC_STRING_01_006: [ If any error occurs, rc_string_create shall fail and return NULL. ]*/
-        LogError("THANDLE_MALLOC_WITH_EXTRA_SIZE(RC_STRING) failed, extra size is %zu, string_length_with_terminator=%zu", sizeof(RC_STRING_INTERNAL) - sizeof(RC_STRING) + string_length_with_terminator, string_length_with_terminator);
+        /* Codes_SRS_RC_STRING_07_010: [ If the resulting memory size requested for the `THANDLE(RC_STRING)`and `string` results in an size_t overflow, `rc_string_create` shall failand return `NULL`. ]*/
+        LogError("Requested memory size overflow.");
```

Comment:
>Error("Requested memory size overflow."); [](http://example.com/codeflow?start=11&length=41)

Let's print in this the string_length at least so that when the error happens we know how much was requested so that we overflowed